### PR TITLE
Aptos nft support

### PR DESCRIPTION
**Pull request by kev1n-peters**

### CODE REVIEW

File: src/components/NFTOriginVerifier.tsx [hunk 1]
```diff
@@ -195,6 +211,56 @@ export default function NFTOriginVerifier() {
           }
         }
       })();
+    } else if (
+      lookupChain === CHAIN_ID_APTOS &&
+      lookupCreatorAddress &&
+      lookupCollectionName &&
+      lookupTokenName
```
Comment (raop155) [inline src/components/NFTOriginVerifier.tsx:218]: Probably we could create something like a `const` boolean on the top to match this validation as:
` const thisIsMySemanticBooleanName:boolean = lookupChain === CHAIN_ID_APTOS && lookupCreatorAddress && lookupCollectionName && lookupTokenName;`

```suggestion
      thisIsMySemanticBooleanName
```

File: src/components/NFTOriginVerifier.tsx [hunk 2]
```diff
@@ -195,6 +211,56 @@ export default function NFTOriginVerifier() {
...
+          if (!cancelled) {
+            setIsLoading(false);
+            setParsedTokenAccount({
+              amount: "0",
+              decimals: 0,
+              mintKey: `${tokenData.collection} ${tokenData.name}`,
+              publicKey: "",
+              uiAmount: 0,
+              uiAmountString: "0",
+              uri: tokenData.uri,
```
Comment (raop155) [inline src/components/NFTOriginVerifier.tsx:252]: This is just a preference, using `destructuring`.

`const {collention, name, uri} = tokenData || {}`

```suggestion
          const tokenData = await tokenClient.getTokenData(
            lookupCreatorAddress,
            lookupCollectionName,
            lookupTokenName
          );
          const {collention, name, uri} = tokenData || {}
          
          if (!cancelled) {
            setIsLoading(false);
            setParsedTokenAccount({
              amount: "0",
              decimals: 0,
              mintKey: `${collection} ${name}`,
              publicKey: "",
              uiAmount: 0,
              uiAmountString: "0",
              uri,
```

File: src/components/NFTOriginVerifier.tsx [hunk 3]
```diff
@@ -206,15 +272,20 @@ export default function NFTOriginVerifier() {
     lookupChain,
     lookupAsset,
     lookupTokenId,
+    lookupCreatorAddress,
+    lookupCollectionName,
+    lookupTokenName,
   ]);
   const readableAddress =
     originInfo &&
```
Comment (raop155) [inline src/components/NFTOriginVerifier.tsx:280]: Probably here also we can use `destructuring` like:

`const {chainId, assetAddress, ....} = originInfo || {}`

This also comes in handy for all `originInfo` properties used in the whole file.

File: src/components/NFTOriginVerifier.tsx [hunk 4]
```diff
@@ -206,15 +272,20 @@ export default function NFTOriginVerifier() {
...
     originInfo.chainId &&
     originInfo.assetAddress &&
-    hexToNativeAssetString(
+    (hexToNativeAssetString(
```
Comment (raop155) [inline src/components/NFTOriginVerifier.tsx:283]: I think this validation could be a `const` with a semantic var name on top of the scope.

File: src/components/NFTOriginVerifier.tsx [hunk 5]
```diff
@@ -245,27 +316,58 @@ export default function NFTOriginVerifier() {
...
+                label="Paste token name"
+                value={lookupTokenName}
+                onChange={handleTokenNameChange}
+              />
+            </>
+          ) : (
+            <>
+              {isEVMChain(lookupChain) ? (
+                <KeyAndBalance chainId={lookupChain} />
+              ) : null}
```
Comment (raop155) [inline src/components/NFTOriginVerifier.tsx:350]: We could use `short-circiut` here, so only when `isEVMChain(lookupChain) ` is truth will display `KeyAndBalance`

```suggestion
              {isEVMChain(lookupChain) &&
                <KeyAndBalance chainId={lookupChain} />}
```

File: src/components/NFTOriginVerifier.tsx [hunk 6]
```diff
@@ -245,27 +316,58 @@ export default function NFTOriginVerifier() {
...
+              <TextField
+                fullWidth
+                variant="outlined"
+                margin="normal"
+                label="Paste an address"
+                value={lookupAsset}
+                onChange={handleAssetChange}
+              />
+              {isEVMChain(lookupChain) ? (
```
Comment (raop155) [inline src/components/NFTOriginVerifier.tsx:359]:
```suggestion
             {isEVMChain(lookupChain) &&
                <KeyAndBalance chainId={lookupChain} />}
```

File: src/components/TokenSelectors/AptosTokenPicker.tsx [hunk 7]
```diff
@@ -16,12 +16,13 @@ type AptosTokenPickerProps = {
...
   disabled: boolean;
   resetAccounts: (() => void) | undefined;
+  nft?: boolean;
 };
 
 const returnsFalse = () => false;
 
 export default function AptosTokenPicker(props: AptosTokenPickerProps) {
-  const { value, onChange, tokenAccounts, disabled } = props;
+  const { value, onChange, tokenAccounts, disabled, nft } = props;
```
Comment (raop155) [inline src/components/TokenSelectors/AptosTokenPicker.tsx:25]: `nft` is optional, so we can do something like this to make `false` as default:
```suggestion
  const { value, onChange, tokenAccounts, disabled, nft = false } = props;
```

File: src/components/TokenSelectors/AptosTokenPicker.tsx [hunk 8]
```diff
@@ -95,9 +96,9 @@ export default function AptosTokenPicker(props: AptosTokenPickerProps) {
 
   const RenderComp = useCallback(
     ({ account }: { account: NFTParsedTokenAccount }) => {
-      return BasicAccountRender(account, returnsFalse, false);
+      return BasicAccountRender(account, returnsFalse, !!nft);
```
Comment (raop155) [inline src/components/TokenSelectors/AptosTokenPicker.tsx:99]: Probably then here you do not have to do a `!!`

File: src/components/TokenSelectors/NFTViewer.tsx [hunk 9]
```diff
@@ -126,6 +128,18 @@ const LogoIcon = ({ chainId }: { chainId: ChainId }) =>
       src={fantomIcon}
       alt="Fantom"
     />
+  ) : chainId === CHAIN_ID_APTOS ? (
```
Comment (raop155) [inline src/components/TokenSelectors/NFTViewer.tsx:131]: Later here in the `LogoIcon` component we can use an `Object / Hashmap` and replace the `ternary` chaining.

File: src/components/TokenSelectors/TokenPicker.tsx [hunk 10]
```diff
@@ -560,14 +560,16 @@ export default function TokenPicker({
...
         </Alert>
-        <TextField
-          variant="outlined"
-          label="Search name or paste address"
-          value={holderString}
-          onChange={(event) => setHolderString(event.target.value)}
-          fullWidth
-          margin="normal"
-        />
+        {nft && chainId === CHAIN_ID_APTOS ? null : (
```
Comment (raop155) [inline src/components/TokenSelectors/TokenPicker.tsx:563]: We could probably create a `const` with this validation

File: src/hooks/useHandleNFTRedeem.tsx [hunk 11]
```diff
@@ -160,13 +165,47 @@ async function solana(
...
+    options?: any
+  ) => Promise<{
+    hash: string;
+  }>
+) {
+  dispatch(setIsRedeeming(true));
+  const nftBridgeAddress = getNFTBridgeAddressForChain(CHAIN_ID_APTOS);
+  try {
+    const msg = await redeemOnAptos(nftBridgeAddress, signedVAA);
+    //msg.arguments[0] = Array.from(msg.arguments[0]);
```
Comment (raop155) [inline src/hooks/useHandleNFTRedeem.tsx:183]: Do we need this comment?

File: src/hooks/useHandleNFTRedeem.tsx [hunk 12]
```diff
@@ -185,7 +224,8 @@ export function useHandleNFTRedeem() {
         solPK.toString(),
         signedVAA
       );
-    } else {
+    } else if (targetChain === CHAIN_ID_APTOS && !!aptosAddress && signedVAA) {
```
Comment (raop155) [inline src/hooks/useHandleNFTRedeem.tsx:227]: Probably we could create a `const` with this validation.

File: src/hooks/useHandleNFTTransfer.tsx [hunk 13]
```diff
@@ -183,6 +192,66 @@ async function solana(
...
+) {
+  dispatch(setIsSending(true));
+  const nftBridgeAddress = getNFTBridgeAddressForChain(CHAIN_ID_APTOS);
+  try {
+    const transferPayload = transferFromAptos(
+      nftBridgeAddress,
+      aptosTokenId.token_data_id.creator,
+      aptosTokenId.token_data_id.collection,
+      aptosTokenId.token_data_id.name,
+      Number(aptosTokenId.property_version),
```
Comment (raop155) [inline src/hooks/useHandleNFTTransfer.tsx:216]: We could use `destructuring` here.

**Review by raop155 — APPROVED** [review]
Looks good to me!
Overall are only suggestions for readability.

File: src/components/NFT/Target.tsx [hunk 14]
```diff
@@ -111,10 +112,10 @@ function Target() {
         fullWidth
         variant="outlined"
         className={classes.transferField}
-        value={readableTargetAddress}
+        value={readableTargetAddress || ""}
         disabled={true}
       />
-      {targetAsset !== ethers.constants.AddressZero ? (
+      {targetAsset && targetAsset !== ethers.constants.AddressZero ? (
```
[the remark this answers is not in the document]
Comment (kev1n-peters) [reply src/components/NFT/Target.tsx]: fixed

File: src/components/NFTOriginVerifier.tsx (hunk 1 above)
Comment (kev1n-peters) [reply src/components/NFTOriginVerifier.tsx]: fixed

File: src/components/NFTOriginVerifier.tsx (hunk 2 above)
Comment (kev1n-peters) [reply src/components/NFTOriginVerifier.tsx]: fixed

File: src/hooks/useHandleNFTTransfer.tsx (hunk 13 above)
Comment (kev1n-peters) [reply src/hooks/useHandleNFTTransfer.tsx]: fixed

File: src/hooks/useHandleNFTRedeem.tsx (hunk 11 above)
Comment (kev1n-peters) [reply src/hooks/useHandleNFTRedeem.tsx]: fixed

File: src/components/NFTOriginVerifier.tsx (hunk 4 above)
Comment (kev1n-peters) [reply src/components/NFTOriginVerifier.tsx]: fixed

File: src/components/NFTOriginVerifier.tsx (hunk 5 above)
Comment (kev1n-peters) [reply src/components/NFTOriginVerifier.tsx]: fixed

**Pull request merged**